parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -0,0 +1,135 @@
+	 */
+	private String authKey;
+
+	public LegionBoardParser(SubstitutionScheduleData scheduleData, CookieProvider cookieProvider) {
+		super(scheduleData, cookieProvider);
+		data = scheduleData.getData();
+	}
+
+	public SubstitutionSchedule getSubstitutionSchedule() throws IOException, JSONException, CredentialInvalidException {
+		api = data.getString("api");
+
+		/*
+		 * QUESTION: Where do I implement the SHA-256 hash of "[username]//[password]"?
+		 * See the documentation: https://gitlab.com/legionboard/heart/blob/master/doc/README.md
+		 */
+		authKey = getCredential().getHash();
+		SubstitutionSchedule v = SubstitutionSchedule.fromData(scheduleData);
+		v.setClasses(getAllClasses());
+		v.setTeachers(getAllTeachers());
+		final JSONArray changes = getChanges();
+		final JSONArray courses = getCourses();
+		final JSONArray teachers = getTeachers();
+
+		parseLegionBoard(changes, courses, teachers);
+		return v;
+	}
+
+	/**
+	 * Returns a JSONArray with all changes from now to in one week.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/changes/list.md">List changes</a>
+	 */
+	private JSONArray getChanges() throws IOException, JSONException {
+		// Date (or alias of date) when the changes start
+		final String startBy = "now";
+		// Date (or alias of date) when the changes end
+		final String endBy = "i1w";
+
+		final String url = api + "/changes?startBy=" + startBy + "&endBy=" + endBy + "&k=" + authKey;
+		return getJSONArray(url);
+	}
+
+	/**
+	 * Returns a JSONArray with all courses.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/courses/list.md">List courses</a>
+	 */
+	private JSONArray getCourses() throws IOException, JSONException {
+		final String url = api + "/courses?k=" + authKey;
+		return getJSONArray(url);
+	}
+
+	/**
+	 * Returns a JSONArray with all teachers.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/teachers/list.md">List teachers</a>
+	 */
+	private JSONArray getTeachers() throws IOException, JSONException {
+		final String url = api + "/teachers?k=" + authKey;
+		return getJSONArray(url);
+	}
+
+	private JSONArray getJSONArray(String url) throws IOException, JSONException {
+		// TODO: get JSONArray from HTML
+		return null;
+	}
+
+	private void parseLegionBoard(JSONArray changes, JSONArray courses, JSONArray teachers) throws IOException {
+		// TODO: Implement (https://gitlab.com/legionboard/kiss/blob/master/index.js)
+	}
+
+	@Override
+	public List<String> getAllClasses() throws JSONException, JSONException {
+		List<String> classes = new ArrayList<>();
+		final JSONArray courses = getCourses();
+		for (JSONArray course : courses) {
+			classes.add(course["name"]);
+		}
+		return classes;
+	}
+
+	@Override
+	public List<String> getAllTeachers() throws IOException, JSONException {
+		List<String> teachers = new ArrayList<>();
+		final JSONArray JSONTeachers = getTeachers();
```

Comment:
just a quick reminder: Variables should start with lowercase letters in Java (see e.g. https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names) -> `jsonTeachers` (or `teachersJson`). :wink: